### PR TITLE
Support SVGElements - Fixes #99

### LINKED ISSUE #99: Can't apply `...attributes` to `SVGElement`
```
app/components/percentage-bar.hbs:5:3 - error TS2344: Type 'Element' does not satisfy the constraint 'SVGElement'.

5   ...attributes
    ~~~~~~~~~~~~~
6 >
```

```hbs
<svg
  width={{this.width}}
  height={{this.height}}
  class="percentage-bar"
  ...attributes
>
  <rect x="0" y="0" width={{this.barWidth}} height={{this.height}}></rect>
</svg>
```

```ts
interface PercentageBarSignature {
  Element: SVGElement;
  Args: {
    width?: number;
    height?: number;
    percentage: number;
    scale?: number;
  };
}
```

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

File: packages/template/-private/dsl/types.d.ts [hunk 1]
```diff
@@ -2,10 +2,15 @@ import { HasContext, HasElement } from '@glint/template/-private/integration';
...
+/**
+ * Given a tag name, returns an appropriate `HTMLOrSVGElement` subtype.
+ * NOTE: This will may not correctly handle SVGElements that have the same name as HTMLElements.
+ */
 export type ElementForTagName<Name extends string> = Name extends keyof HTMLElementTagNameMap
   ? HTMLElementTagNameMap[Name]
-  : Element;
+  : Name extends keyof SVGElementTagNameMap
+  ? SVGElementTagNameMap[Name]
+  : HTMLOrSVGElement;
```
[the remark this answers is not in the document]
Comment (dfreeman) [reply packages/template/-private/dsl/types.d.ts:13]: Regardless of any of that, I don't think `HTMLOrSVGElement` is actually what we want here. Contrary to the name, it's not equivalent to `HTMLElement | SVGElement`, but actually a super small common interface they both implement that's missing a lot of the important actual `Element` API:

```ts
interface HTMLOrSVGElement {
    autofocus: boolean;
    readonly dataset: DOMStringMap;
    nonce?: string;
    tabIndex: number;
    blur(): void;
    focus(options?: FocusOptions): void;
}
```

Given how Glimmer behaves, I think we probably want `HTMLElement | SVGElement` as the fallback value here.

File: packages/template/-private/dsl/invoke.d.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export declare function yieldToBlock<Context extends AnyContext, K extends keyof
  *     applySplattributes<typeof 𝚪.element, ElementForComponent<typeof AnotherComponent>>();
  */
 export declare function applySplattributes<
-  SourceElement extends Element,
+  SourceElement extends HTMLOrSVGElement,
```
Comment (dfreeman) [inline packages/template/-private/dsl/invoke.d.ts:69]: See below, but I think `Element` is correct here (`SVGOrHTMLElement` knows nothing about attributes)

File: packages/template/-private/dsl/types.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (wagenet) [reply packages/template/-private/dsl/types.d.ts:13]: Makes sense. I think we do want the SVG tag lookup, but still base type of Element.

File: packages/template/-private/dsl/types.d.ts [hunk 3]
```diff
@@ -2,9 +2,21 @@ import { HasContext, HasElement } from '@glint/template/-private/integration';
...
+
+/**
+ * Given a tag name, returns an appropriate `Element` subtype.
+ * NOTE: This will return a union for elements that exist both in HTML and SVG. Technically, this will be too permissive.
+ */
+export type ElementForTagName<Name extends string> = Name extends
+  | keyof HTMLElementTagNameMap
+  | keyof SVGElementTagNameMap
+  ? LookupHTMLElement<Name> | LookupSVGElement<Name>
   : Element;
```
Comment (wagenet) [inline packages/template/-private/dsl/types.d.ts:20]: Just realized I didn't change this to `HTMLElement | SVGElement` should I still do that?

**Review by dfreeman — APPROVED** [review]

**Review by dfreeman — APPROVED** [review]

**Pull request merged**